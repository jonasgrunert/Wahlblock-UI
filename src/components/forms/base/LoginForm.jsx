import * as React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxField } from 'redux-form';
import { Button, Columns, Column, Control, Field, Input as BInput, Label } from 'bloomer';

export const TextInput = field => <BInput {...field.input} />;

TextInput.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  state: PropTypes.bool,
};

TextInput.defaultProps = {
  state: false,
};

export class LoginForm extends React.Component {
  componentWillMount() {
    this.props.generateKey();
  }

  render() {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <Field>
          <Label>Name</Label>
          <Control>
            <ReduxField
              name="name"
              component={TextInput}
              type="text"
              placeholder="Max Mustermann"
            />
          </Control>
        </Field>
        <Field>
          <Label>PIN</Label>
          <Control>
            <ReduxField
              name="pin"
              component={TextInput}
              type="password"
              placeholder="Your PIN"
              state={this.props.state}
            />
          </Control>
        </Field>
        <Columns isMobile>
          <Column isSize="full">
            <Button
              type="submit"
              isColor="primary"
              isSize="medium"
              isPulled="right"
              isOutlined
              isLoading={this.props.state}
            >
              Login
            </Button>
          </Column>
        </Columns>
      </form>
    );
  }
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  generateKey: PropTypes.func.isRequired,
  state: PropTypes.bool,
};

LoginForm.defaultProps = {
  state: false,
};
