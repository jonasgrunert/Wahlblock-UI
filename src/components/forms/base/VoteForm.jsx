import * as React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxField } from 'redux-form';
import { Button, Control, Column, Columns, Field, Label, Radio } from 'bloomer';

const Option = field => <Radio {...field.input}> {field.input.value} </Radio>;

const Options = optionsConfig =>
  optionsConfig.map(config => (
    <span>
      <ReduxField
        name="selection"
        component={Option}
        type="radio"
        option={config.option}
        value={config.option}
      />
      <br />
    </span>
  ));

export const Vote = props => (
  <form onSubmit={props.handleSubmit}>
    <Field>
      <Label hasTextAlign="left">Your Selection</Label>
      <Control>{Options(props.options)}</Control>
    </Field>
    <Columns isMobile>
      <Column isSize="full">
        <Button
          type="submit"
          isColor="primary"
          isSize="medium"
          isPulled="right"
          isLoading={props.state}
        >
          Vote
        </Button>
      </Column>
    </Columns>
  </form>
);

Vote.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  state: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.string),
};

Vote.defaultProps = {
  state: false,
  options: ['Ja', 'Nein'],
};
