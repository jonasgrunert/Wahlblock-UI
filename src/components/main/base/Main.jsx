import { Box, Column, Columns, Container, Notification, Title } from 'bloomer';
import * as React from 'react';
import PropTypes from 'prop-types';

import LoginFormContainer from '../../forms/hoc/loginForm';

export class Main extends React.Component {
  componentDidMount() {
    this.props.redirect();
  }

  componentDidUpdate() {
    this.props.redirect();
  }

  render() {
    return (
      <Container hasTextAlign="centered">
        <Columns isCentered>
          <Column isSize="1/2">
            <Box hasTextAlign="left">
              <Title hasTextColor="primary">Login</Title>
              <Notification isColor="danger">
                Once logged in on one device you have to complete the vote process.
              </Notification>
              <LoginFormContainer />
            </Box>
          </Column>
        </Columns>
      </Container>
    );
  }
}

Main.propTypes = {
  redirect: PropTypes.func.isRequired,
};
