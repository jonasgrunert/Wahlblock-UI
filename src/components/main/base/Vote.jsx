import { Box, Container, Column, Columns, Title } from 'bloomer';
import * as React from 'react';
import PropTypes from 'prop-types';

import VoteFormWrapper from '../../forms/hoc/voteForm';

// STateful compoent for redirecting upon login
export class Vote extends React.Component {
  componentWillMount() {
    this.props.redirect();
  }

  componentWillUpdate() {
    this.props.redirect();
  }

  render() {
    return (
      <Container hasTextAlign="centered">
        <Columns isCentered>
          <Column isSize="1/2">
            <Box>
              <Title isSize={2} hasTextColor="black">
                Vote now
              </Title>
              <VoteFormWrapper />
            </Box>
          </Column>
        </Columns>
      </Container>
    );
  }
}

Vote.propTypes = {
  redirect: PropTypes.func.isRequired,
};
