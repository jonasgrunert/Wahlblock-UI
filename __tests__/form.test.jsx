import * as React from 'react';
import { LoginForm, TextInput } from '../src/components/forms/base/LoginForm';
// import { Vote } from '../src/components/forms/voteForm';

describe('rendering forms', () => {
  it('render textinput', () => {
    const textinput = TextInput({ input: 'text' });
    expect(textinput).toMatchSnapshot();
  });
  it('render loginform', () => {
    const login = shallow(<LoginForm generateKey={jest.fn()} handleSubmit={jest.fn()} />);
    expect(login).toMatchSnapshot();
  });
  // it('render voteform', () => {
  //   const vote = shallow(<Vote handleSubmit={jest.fn()} />);
  //   expect(vote).toMatchSnapshot();
  // });
});
