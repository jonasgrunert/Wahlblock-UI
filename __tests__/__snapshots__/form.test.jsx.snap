// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering forms render loginform 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <withHelpersModifiers(Field)>
    <withHelpersModifiers(Label)>
      Name
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Control)>
      <Field
        component={[Function]}
        name="name"
        placeholder="Max Mustermann"
        type="text"
      />
    </withHelpersModifiers(Control)>
  </withHelpersModifiers(Field)>
  <withHelpersModifiers(Field)>
    <withHelpersModifiers(Label)>
      PIN
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Control)>
      <Field
        component={[Function]}
        name="pin"
        placeholder="Your PIN"
        state={false}
        type="password"
      />
    </withHelpersModifiers(Control)>
  </withHelpersModifiers(Field)>
  <withHelpersModifiers(Columns)
    isMobile={true}
  >
    <withHelpersModifiers(Column)
      isSize="full"
    >
      <withHelpersModifiers(Button)
        isColor="primary"
        isLoading={false}
        isOutlined={true}
        isPulled="right"
        isSize="medium"
        type="submit"
      >
        Login
      </withHelpersModifiers(Button)>
    </withHelpersModifiers(Column)>
  </withHelpersModifiers(Columns)>
</form>
`;

exports[`rendering forms render textinput 1`] = `
<withHelpersModifiers(Input)
  0="t"
  1="e"
  2="x"
  3="t"
/>
`;
